staging/src/k8s.io/client-go/rest/url_utils.go

```diff
@@ -84,7 +84,7 @@ func defaultServerUrlFor(config *Config) (*url.URL, string, error) {
 	// config.Insecure is taken to mean "I want HTTPS but don't bother checking the certs against a CA."
 	hasCA := len(config.CAFile) != 0 || len(config.CAData) != 0
 	hasCert := len(config.CertFile) != 0 || len(config.CertData) != 0
-	defaultTLS := hasCA || hasCert || config.Insecure
```

Comment:
`config.Insecure = true` here means "use https but do not verify the certificate", not "use http"

Reply:
Okey, get it, thanks :)